components/Navbar/index.js

```diff
@@ -60,6 +64,38 @@ const Navbar = () => {
   const { isLogin } = useSelector((state) => state.sign);
   const { totalItems } = useSelector((state) => state.cart);
 
+  //use session login user
+  const dispatch = useDispatch();
+  const { data: session, error } = useSession();
+
+  useEffect(()=>{
+    if (session && !isLogin) {
+      //extract data from session token
+      const provider = session.token.account.provider;
+      const providerAccountId = session.token.account.providerAccountId;
+      const username = session.token.user.name;
+      const email = session.token.user.email;
+      const avatarUrl = session.token.user.image;
+      //dispatch loginAction
+      dispatch(
```

Comment:
The issue that you need to click 'logout' twice is because after click the first 'logout', the page will re-render itself. From the useEffect hook here, it will automatically login again from the non-cleared session. 

Once the session is cleared, that you click the 'logout' second time, since the session was cleared by your first clicked. useEffect here won't trigger OAuthLogin again.

So to cope this problem, one way is trying to add dependencies to useEffect and add further marginal conditions.

Reply:
BTW I can't use normal signup and login function (aside from OAuth) from your frontend and backend pr, please tell me if it is the problem only existing on my local machine.